pkg/sidecar/attributes.go

```diff
@@ -0,0 +1,106 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package sidecar contains operations related to sidecar manipulation (Add, update, remove).
+package sidecar
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const resourceAttributesEnvName = "OTEL_RESOURCE_ATTRIBUTES"
+
+type podReferences struct {
+	replicaset *appsv1.ReplicaSet
+	deployment *appsv1.Deployment
+}
+
+// getAttributesEnv returns a list of environment variables. This consists of some pod fields and an environment variable OTEL_RESOURCE_ATTRIBUTES which contains a set of different pod references.
+// see: https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/
+func getAttributesEnv(ns corev1.Namespace, podReferences podReferences) []corev1.EnvVar {
```

Comment:
I would incorporate resource(s) in the name because there are two types of attributes - resource and span.